.github/workflows/ci.yml

```diff
@@ -6,51 +6,33 @@ on:
       - master
   pull_request:
 
-jobs:
-  check:
-    name: Check
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - run: |
-          rustup override set stable
-          rustup update stable
-      - uses: Swatinem/rust-cache@v1.3.0
-      - run: cargo check
-        env:
-          RUSTFLAGS: "-D warnings"
+env:
+  # minimum supported rust version
+  MSRV: 1.51.0
 
+jobs:
   tests:
     name: Tests
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - run: |
-          rustup override set stable
-          rustup update stable
+          rustup override set ${{ env.MSRV }}
+          rustup update ${{ env.MSRV }}
       - uses: Swatinem/rust-cache@v1.3.0
       - run: cargo test
+        env:
+          RUSTFLAGS: "-D warnings"
 
-  fmt:
-    name: Rustfmt
+  lints:
+    name: Rustfmt and Clippy
```

Comment:
any reason for merging these together?

Reply:
No strong reason, do you prefer to split them?

Reply:
yeah. mainly because if rustfmt fails then you wouldn't see the issues that clippy reports.

similarly for the `cargo check` and `cargo test` merge above. for this repo it doesn't matter too much, but `cargo test` does not compile the examples by default, which is why I usually have `cargo check` as a dedicated job so that **all** targets are checked for compilability.

Reply:
Makes sense, dropped the (previous) second commit.